### PR TITLE
Pipeline: support insert event

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### What is changed and how it works?
- add `insertEvent` to dynamically adjust the topology of events.
```
    /// eventA───────►eventB ===> eventA─────────────►eventB
    ///                             │                    ▲
    ///                             └────►insert_event───┘
    void insertEvent(const EventPtr & insert_event) noexcept;
```
- refine some logs like #7285

### Check List

https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/1018/pipeline
`TEST_F(EventTestRunner, insert_events)` tsan passed

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp [hunk 1]
```diff
@@ -125,6 +140,7 @@ void Event::finish() noexcept
     // If query has already been cancelled, it will not trigger outputs.
     if (likely(!exec_status.isCancelled()))
     {
+        std::cout << fmt::format("schedule {} outputs", outputs.size()) << std::endl;
```
Comment (SeaRise) [inline dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp:143]:
```suggestion
```

**Comment by SeaRise** [comment]
> > /run-sanitizer-test tsan
> 
> https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/1018/pipeline

`TEST_F(EventTestRunner, insert_events)` passed

File: dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp [hunk 2]
```diff
@@ -41,17 +41,36 @@ extern const char random_pipeline_model_event_finish_failpoint[];
...
+    assert(input.get() != this);
     input->addOutput(shared_from_this());
     ++unfinished_inputs;
     is_source = false;
 }
 
 void Event::addOutput(const EventPtr & output)
 {
-    assert(status == EventStatus::INIT);
+    /// Output will also be added in the Finish state, as can be seen in the `insertEvent`.
```
Comment (SeaRise) [inline dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp:52]:
```suggestion
    /// Output will also be added in the Finished state, as can be seen in the `insertEvent`.
```

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**